include/ultra64/gbi.h

```diff
@@ -1430,120 +1775,155 @@ typedef struct {
  * } Gsegment;
  */
 typedef struct {
-        int     cmd:8;
-        int     pad0:8;
-        int     mw_index:8;
-        int     number:8;
-        int     pad1:8;
-        int     base:24;
+    int cmd      : 8;
+    int pad0     : 8;
+    int mw_index : 8;
+    int number   : 8;
+    int pad1     : 8;
+    int base     : 24;
 } Gsegment;
 
 typedef struct {
-        int     cmd:8;
-        int     pad0:8;
-        int     sft:8;
-        int     len:8;
-        unsigned int    data:32;
+    int          cmd  : 8;
+    int          pad0 : 8;
+    unsigned int sft  : 8;
+    unsigned int len  : 8;
+    unsigned int data : 32;
 } GsetothermodeL;
 
 typedef struct {
-        int     cmd:8;
-        int     pad0:8;
-        int     sft:8;
-        int     len:8;
-        unsigned int    data:32;
+    int          cmd  : 8;
+    int          pad0 : 8;
+    unsigned int sft  : 8;
+    unsigned int len  : 8;
+    unsigned int data : 32;
 } GsetothermodeH;
 
 typedef struct {
-        unsigned char   cmd;
-        unsigned char   lodscale;
-        unsigned char   tile;
-        unsigned char   on;
-        unsigned short  s;
-        unsigned short  t;
+    unsigned char  cmd;
+    unsigned char  lodscale;
+#ifdef F3DEX_GBI_2
+    unsigned char  pad   : 2;
+    unsigned char  level : 3;
+    unsigned char  tile  : 3;
+#else
+    unsigned char  tile;
+#endif
+    unsigned char  on;
```

Comment:
I'm not sure if it's totally worth going down the rabbit hole of making these structs extremely accurate, especially considering that based on the ucode_disas codegen we have to implement this sort of matching trick. Not even the original writers of gbi.h bothered to update these structs for F3DEX(2) 😅

Reply:
I feel like if we're going to edit gbi.h we should try to do it less sloppily than what it already is, though I agree it doesn't really matter...

Reply:
It's more so that we have to comply with ucode_disas codegen and this sort of union hack to both fulfill codegen needs and bit accuracy is imo not really worth it